tests/cpp_integration_tests/test_main_model_static.cpp

```diff
@@ -29,9 +29,9 @@ TEST_CASE("Test main model static") {
     update_data_independent.add_buffer("link", -1, link_indptr.crbegin()[1], link_indptr.data(), source.data());
     update_data_independent.add_buffer("source", -1, source_indptr.crbegin()[1], source_indptr.data(), source.data());
 
-    SUBCASE("Independent update data") { CHECK(MainModel::is_update_independent(update_data_independent) == true); }
+    // SUBCASE("Independent update data") { CHECK(MainModel::is_update_independent(update_data_independent) == true); }
 
-    SUBCASE("Dependent update data") { CHECK(MainModel::is_update_independent(update_data_dependent) == false); }
+    // SUBCASE("Dependent update data") { CHECK(MainModel::is_update_independent(update_data_dependent) == false); }
```

Comment:
These two tests are no longer possible due to the fact that function `is_update_independent` now is not `static` and the checking of update data requires the reference to input data. Probably need to either remove this test source file completely, or still maintain this by add dedicated logic as `static`.

Reply:
Is there a TODO to be added here?
Or removing would do.

Reply:
@TonyXiang8787 I'd prefer removing but it might depend on how our users would want to use this? i.e., if they really have to have it as static

Reply:
* i'm for removing it if it is no longer used.
* if they are still used, they don't have to be static, but please make sure that it has as few dependencies on the main model as possible (and test if public).

the only reason this was static was because it only depended on update data alone. since you now also require input data (because of ID mapping), you can do one of 2 things:
* [ ] provide both input and update data as input parameters (go ahead; no discussion needed).
  * this keeps it static (which makes it trivial to move out of main model; something you can also do in the follow-up refactoring we've already talked about)
* or make `const`, not `static`
  * also OK with that, but be careful. The danger is that refactoring becomes harder if you want to move some checks to `main_core`, as we discussed. the moment something is not `static` in a large class, you basically have access to "everything" and it becomes very tempting to actually use "everything". that's a recipe for "organically growing code". since that's precisely what we're struggling with at the moment, be careful if you decide to go down this path.

Reply:
I propose to make it non-static at this moment. The refactor will be a big thing, this choice will be minor for the whole stuff.

Reply:
:+1: agree with this as long as we revisit this in the near future. let's try to remember re-adding the unit tests

Reply:
Wow we can edit checkboxes in other people's comment XD

Reply:
As discussed, conversation resolved.